sdk/ai/ai-projects/review/ai-projects.api.md

```diff
@@ -0,0 +1,1988 @@
+## API Report File for "@azure/ai-projects"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AbortSignalLike } from '@azure/abort-controller';
+import { ClientOptions } from '@azure-rest/core-client';
+import { Paged } from '@azure/core-paging';
+import { RequestParameters } from '@azure-rest/core-client';
+import { StreamableMethod } from '@azure-rest/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface AgentDeletionStatusOutput {
+    deleted: boolean;
+    id: string;
+    object: "assistant.deleted";
+}
+
+// @public
+export interface AgentEventMessage {
+    data: AgentThreadOutput | ThreadRunOutput | RunStepOutput | ThreadMessageOutput | MessageDeltaChunk | RunStepDeltaChunk | string;
+    event: AgentStreamEventType | string;
+}
+
+// @public
+export interface AgentEventMessageStream extends AsyncDisposable, AsyncIterable<AgentEventMessage> {
+}
+
+// @public
+export interface AgentOutput {
+    created_at: number;
+    description: string | null;
```

Comment:
We usually do not use `null`, preferring `undefined` - is this really returned as `null` on the wire? If not, maybe use `description?: string`

Same for the rest of the parameters

Reply:
These are autogenerated from typespec

Reply:
Seems like we have a lot of suppressions of the `no-nullable` rule in the TSP: https://github.com/Azure/azure-rest-api-specs/blob/e0249a61a6b884891b68d0224350624378b26f76/specification/ai/Azure.AI.Projects/agents/models.tsp#L131

Not sure if we have a good way to fix this in codegen, might be something that has to be hand-tooled given the other naming limitations

Reply:
@ganeshyb so on the wire the property `description` could be a JSON value `null`? if we want to distinct nullability and optionality on the wire, i was wondering if client customers need to distinct them also?